### PR TITLE
add VSCode and CMake instructions

### PR DESCRIPTION
**Pull request by danewalton**
Adding a `settings.json` file to the `.vscode` directory. This integrates with the CMake extension by Microsoft and will help users debug and run samples and tests.
- The file is ignored in the `.gitignore` so if they paste SAS keys or other things in there, it won't get pushed in a branch.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "cmake.configureEnvironment": {
```
Comment (vhvb1989) [inline .vscode/settings.json:2]: ah, Feel good to see this kind of vscode PRs!

I tried to add launch.json and tasks.json in the past so customers could already have debugging env ready to launch debugger with vscode.

But the team didn't see it as a good idea to add this kind of utilities.

As a customer, I like when I am given all of this things that everyone needs to eventually figure it out

Comment (danewalton) [reply .vscode/settings.json:2]: Yea the only nit I have with the build.json and tasks.json is that they are VERY involved and in some ways require a different lesson in how to build the SDK. Doing it this way, the user just sets various options and the extension handles it for them. I think it's overall an easier system to maintain and easier for the user to use.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    "VCPKG_ROOT": "",
+    "VCPKG_DEFAULT_TRIPLET": ""
+  },
+  "cmake.configureSettings": {
+    "WARNINGS_AS_ERRORS" : "ON",
+    "TRANSPORT_CURL" : "OFF",
+    "UNIT_TESTING" : "OFF",
```
Comment (ahsonkhan) [inline .vscode/settings.json:9]: > will help users debug and run samples and tests.

If the primary benefit here is to debug and run samples/tests, unit testing should be "ON", no?

**Comment by ahsonkhan** [comment]
> - The file is ignored in the `.gitignore` so if they paste SAS keys or other things in there, it won't get pushed in a branch.

I don't think the file is being ignored. I made changes to my .vscode/settings file and they are showing up in git diff:
image

Maybe we need to add the following to the `.gitignore` file, or find some other way to make sure folks aren't accidentally checking in anything within their local `.vscode` folder.
> .vscode/*.*

https://github.com/Azure/azure-sdk-for-c/blob/e33b40000cc8593d3eb8570d3f3970bcfd129dc9/.gitignore#L307-L308

That was precisely the reason why we added it to the `.gitignore` to begin with and why checking in such files (build, task, settings json, etc.) was considered problematic.

@danewalton, can you please verify that changes to such files are actually ignored by git, and otherwise make any of the necessary `.gitignore` changes to make that happen?

**Comment by danewalton** [comment]
@ahsonkhan yea we have an issue here that I think Cris opened an issue for. We might have to create some `vscode-config` folder and put the files in there where people can then copy over the `settings.json` file to their `.vscode`

File: README.md [hunk 3]
```diff
@@ -200,6 +201,14 @@ The following CMake options are available for adding/removing project features.
 
       i.e. cmake -DTRANSPORT_CURL=ON ..
 
+### VSCode
+
+For convenience, you can quickly get started using [VSCode](https://code.visualstudio.com/) and the [CMake Extension by Microsoft](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cmake-tools&ssr=false#overview). Included in the repo is a `settings.json` file [here](https://github.com/Azure/azure-sdk-for-c/blob/master/.vscode/settings.json) which the extension will use to configure a CMake project. With this, you can run and debug samples and tests. Modify the variables in the file to your liking or as instructed by sample documentation and then select the following button in the extension:
+
+![VSCode CMake Config](./sdk/docs/resources/vscode_cmake_config.png)
```
Comment (ahsonkhan) [inline README.md:208]: Why did we choose `sdk/docs/resources/` for this image, when the README is a the root?

Should we move it to `<root of repo>/resources`?

**Pull request merged**